scheduling/tasks_test.go

```diff
@@ -0,0 +1,148 @@
+package scheduling
+
+import (
+	"testing"
+	"time"
+)
+
+func TestSerializeTaskMetadata(t *testing.T) {
+	testCases := []struct {
+		name string
+		task TaskMetadata
+	}{
+		{
+			name: "WithProviderTarget",
+			task: TaskMetadata{
+				ID:   1,
+				Name: "provider_task",
+				Type: HealthCheck,
+				Target: Provider{
+					Name:       "postgres",
+					TargetType: ProviderTarget,
+				},
+				Date: time.Now(),
+			},
+		},
+		{
+			name: "WithNameVariantTarget",
+			task: TaskMetadata{
+				ID:   1,
+				Name: "nv_task",
+				Type: ResourceCreation,
+				Target: NameVariant{
+					Name:       "transaction",
+					TargetType: NameVariantTarget,
+				},
+				Date: time.Now(),
+			},
+		},
+	}
+
+	for _, currTest := range testCases {
+		t.Run(currTest.name, func(t *testing.T) {
+			serializeTask, err := currTest.task.ToJSON()
+			if err != nil {
+				t.Errorf("failed to serialize task metadata: %v", err)
+			}
+
+			deserializeTask := TaskMetadata{}
+			if err := deserializeTask.FromJSON(serializeTask); err != nil {
+				t.Errorf("failed to deserialize task metadata: %v", err)
+			}
+
+			if !TaskMetadataIsEqual(deserializeTask, currTest.task) {
```

Comment:
you should be able to use reflect.DeepEquals here instead so we dont have to change the function below if the struct is updated